runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -375,9 +411,7 @@ function TSHighlighter._on_spell_nav(_, _, buf, srow, _, erow, _)
   local highlight_states = self._highlight_states
   self:prepare_highlight_states(srow, erow)
 
-  for row = srow, erow do
-    on_line_impl(self, buf, row, true)
-  end
+  on_range_impl(self, buf, srow, 0, erow, 0, true)
```

Comment:
Why did the previous implementation include the `erow` line?

`self:prepare_highlight_states` above expects end line to be exclusive.